src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1488,18 +1488,23 @@ private void CloseConnection()
             var stdErrReader = _stdErrReader;
             if (stdErrReader != null) { stdErrReader.Dispose(); }
 
-            try
+            var sshProcessId = _sshProcessId;
```

Comment:
Can't you use `_sshProcessId` directly instead of creating a new variable?

Reply:
CloseConnection can be called from multiple places (state machine, error handling) and this is just to prevent going through this code path multiple times.  I am not concerned about race conditions since Process type is thread safe ... I just want to prevent running the code path after the process has been terminated.

I'll add a comment to clarify.